src/PiggyCustomEnchants/EventListener.php

```diff
@@ -1018,9 +1022,40 @@ public function checkArmorEnchants(Entity $entity, Event $event)
                     if ($enchantment !== null) {
                         if ($event->getDamage() >= $entity->getHealth()) {
                             if ($enchantment->getLevel() > 1) {
-                                $entity->getInventory()->setArmorItem($slot, $this->plugin->addEnchantment($armor, $enchantment->getId(), $enchantment->getLevel() - 1));
-                            } else {
-                                $entity->getInventory()->setArmorItem($slot, $this->plugin->removeEnchantment($armor, $enchantment));
+                                switch($slot){
```

Comment:
Why?

Reply:
`setArmorItem()` was removed from `PlayerInventory` in commit 
https://github.com/pmmp/PocketMine-MP/commit/6543d969102b6ce761717073fdd858467217b876#diff-d32fe31d118541ac4628b565b81ae7f1L236

In this commit, ArmorInventory was created as its own class with no function comparable to `setArmorItem()`.  Each armor position has to be set with separate calls, thus the change to the switch statement.  Let me know if you would prefer this be handled a different way.